tests/integration/mount_propagation.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+  teardown_busybox
+  setup_busybox
+
+  # Create a "shared" parent mount point for container
+  # rootfs and make sure pivot_root() still works. This
+  cd ${BUSYBOX_BUNDLE}
+  mount --bind . .
```

Comment:
@cyphar, not sure how to use `unshare -m` in this script. IIUC, I will have to have another script to run actual test and do `unshare -m <new-script>`. If I just call `unshare -m` it will just launch another bash in new mount namespace but that does not run my rest of the test code.